elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -86,4 +87,16 @@ default void flush() {
         // default to ignoring criteria
         return loadObjects(entityClass);
     }
+
+    /**
+     * Filter a collection by the Predicates in filterScope.
+     *
+     * @param collection the collection to filter
+     * @param type
+     * @param filterScope the FilterScope containing the set of Predicates
+     * @return the filtered collection
+     */
+    default Collection filterCollection(Collection collection, String type, FilterScope<?> filterScope) {
```

Comment:
I don't feel like 'type' belongs here.  The collection already has the type contained within it.  All you need to do is something like:
        if (collection.getClass() instanceof ParameterizedType) {
            return (Class<?>) ((ParameterizedType) collection.getClass()).getActualTypeArguments()[0];
        }

In general, having two parameters that indirectly contain the same information can lead to conflicts where it is possible to specify different information and lead to subtle bugs in the calling code.